src/friction-compensation.cpp

```diff
@@ -0,0 +1,37 @@
+#include "simple-mpc/friction-compensation.hpp"
+
+namespace simple_mpc
+{
+
+  FrictionCompensation::FrictionCompensation(const Model & model, const long actuation_size)
+  {
+    corrected_torque_.resize(actuation_size);
+    dry_friction_ = model.friction;
+    viscuous_friction_ = model.damping;
+  }
+
+  void FrictionCompensation::computeFriction(const Eigen::VectorXd & velocity, const Eigen::VectorXd & torque)
+  {
+    if (velocity.size() != corrected_torque_.size())
+    {
+      throw std::runtime_error("Velocity has wrong size");
+    }
+    if (torque.size() != corrected_torque_.size())
+    {
+      throw std::runtime_error("Torque has wrong size");
+    }
+    corrected_torque_ = torque + viscuous_friction_.cwiseProduct(velocity)
+                        + dry_friction_.cwiseProduct(velocity.unaryExpr(std::function(signFunction)));
+  }
+
+  double FrictionCompensation::signFunction(double x)
```

Comment:
I think there already a wide egein operator to do the sign function. I don't think we have to re-implemt it.
(let's just check if we want (-1,1) or (-1,0,1)).